src/js/providers/tracks-mixin.ts

```diff
@@ -310,8 +310,13 @@ const Tracks: TracksMixin = {
 
             for (i; i < len; i++) {
                 const track = tracks[i] as TextTrackLike;
-                let trackId: string = track._id || '';
-                if (!trackId) {
+                let trackId: string = track._id || '';                
+
+                if (disableIfOldCaptionsSafari(_tracksById, track, trackId)) {
+                    continue;
+                }
+
+                if (!trackId) {   
```

Comment:
What we want to do it prevent `track.inuse` from being set back to `true` for captions tracks that were previously used for native rendering which have `track.inuse` currently set to `false`. This should happen in the `if (!trackId) {` block.